templates/yaml_templates_generator.py

```diff
@@ -115,25 +85,111 @@ def yaml_generator(path_to_modules: str, output_folder: str) -> int:
                             append = False
                             break
                     if append:
-                        if str(v).find("=") == -1 and str(k) != "kwargs":
-                            params_dict[str(k)] = "REQUIRED"
-                        elif str(k) == "kwargs":
-                            params_dict["#additional parameters"] = "AVAILABLE"
-                        else:
-                            params_dict[str(k)] = v.default
+                        _set_param_value(k, v, params_dict)
+            method_dict = {
+                "method": method_name,
+                "module_path": module_name,
+                "parameters": params_dict,
+            }
+            _set_dict_special_cases(method_dict, method_name)
+            params_list = [method_dict]
+            _save_yaml(module_name, method_name, params_list)
+    return 0
 
-            params_list = [{module_name: {method_name: params_dict}}]
 
-            # save the list as a YAML file
-            path_dir = output_folder + "/" + module_name
-            path_file = path_dir + "/" + str(method_name) + ".yaml"
+def _set_param_value(k: int, v: int, params_dict: Dict):
+    """Set param value for method inside dictionary
+    Args:
+        k: Method name, dict key
+        v: Method value, dict value
+        params_dict: Parameter dictionary
+    """
+    if str(v).find("=") == -1 and str(k) != "kwargs":
+        params_dict[str(k)] = "REQUIRED"
+    elif str(k) == "kwargs":
+        params_dict["#additional parameters"] = "AVAILABLE"
+    elif str(k) == "center":
+        # Temporary value
+        params_dict[str(k)] = "$ {{centering.side_outputs.centre_of_rotation}}"
```

Comment:
Tiny nitpick: github action YAML syntax has no space between the dollar sign and the curly braces when making references to other values (similar to substitution in sh/bash) https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-using-a-single-dimension-matrix, so I think this should maybe be changed from
```
"$ {{centering.side_outputs.centre_of_rotation}}"
```

to
```
"${{centering.side_outputs.centre_of_rotation}}"
```

That change will probably make the parsing of the string a bit easier (ie, detecting the `${{}}`) when it comes to it, leaving the space in there could make things a bit more difficult that it needs to be?

Reply:
Thank you for reading through and picking this up

Reply:
No problem, was happy to read through it. The YAML template generation logic separated into a few more functions looks good :slightly_smiling_face: